adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDD.scala

```diff
@@ -85,6 +86,32 @@ sealed trait AlignmentRecordRDD extends AvroReadGroupGenomicRDD[AlignmentRecord,
   }
 
   /**
+   * Converts this set of reads into Coverage records
+   *
+   * @return Returns a CoverageRDD where all reads have been grouped and
+   * counted at all overlapping positions
+   */
+  def toCoverage(collapse: Boolean = true): CoverageRDD = {
+    val coverage: RDD[Coverage] =
+      rdd.rdd
+        .flatMap(r => {
+          val t: List[Long] = List.range(r.getStart, r.getEnd)
+          t.map(n => (ReferenceRegion(r.getContigName, n, n + 1), 1))
+        }).reduceByKey(_ + _)
+        .cache()
```

Comment:
You don't want the `.cache()` call here. I mean, you do, but you don't want the code after it. This should be broken up:

```
val covCounts: RDD[Coverage] =
       rdd.rdd
         .flatMap(r => {
           val t: List[Long] = List.range(r.getStart, r.getEnd)
           t.map(n => (ReferenceRegion(r.getContigName, n, n + 1), 1))
         }).reduceByKey(_ + _)
         .cache()

val coverage = covCounts.sortByKey()
        .map(r => Coverage(r._1, r._2.toDouble))
```